docs/src/modules/components/DemoSandbox.js

```diff
@@ -170,21 +214,51 @@ function DemoSandbox(props) {
 
   return (
     <DemoErrorBoundary name={name} onResetDemoClick={onResetDemoClick} t={t}>
-      {usesCssVarsTheme ? (
-        children
+      {isolated ? (
+        // Place ThemeProvider from MUI System here to disconnect the theme inheritance for Material UI and Joy UI
+        // The demo will need to handle the ThemeProvider itself.
+        <SystemThemeProvider
+          theme={(upperTheme) => ({
+            direction: upperTheme.direction, // required for internal ThemeProvider
+            vars: upperTheme.vars, // required for styling Iframe
+          })}
+        >
+          {iframe ? (
+            <DemoIframe name={name} isJoy={isJoy} isolated={isolated} {...other}>
+              {/* `children` needs to be a child of `DemoIframe` since the iframe implementation rely on `cloneElement`. */}
+              {/* the `colorSchemeNode` will be provided by DemoIframe through `window` prop */}
+              <IsolatedDemo cssVarPrefix={name}>{children}</IsolatedDemo>
+            </DemoIframe>
+          ) : (
+            <IsolatedDemo cssVarPrefix={name} colorSchemeNode={root}>
+              {children}
+            </IsolatedDemo>
+          )}
+        </SystemThemeProvider>
```

Comment:
@alexfauquette The new `isolated` flag is to support these kind of demos:

- [Toolpad demo](https://deploy-preview-45661--material-ui.netlify.app/material-ui/react-drawer/#dashboard-layout) on the drawer page
- [Joy UI tutorial](https://deploy-preview-45661--material-ui.netlify.app/joy-ui/getting-started/tutorial/#interactive-demo)

These are considered mini apps within the docs, so they have to be isolated from the page's theme and color mode.

https://github.com/user-attachments/assets/bf515bad-26b8-4df7-84ec-1568331f7e47